src/utils/replaceTokens.js

```diff
@@ -0,0 +1,54 @@
+
+/**
+ * Prototype code is at:
+ * https://codereview.stackexchange.com/questions/159469/helper-functions-to-perform-simple-token-replacements-in-a-string
+ */
+
+//  Token replacer function"
+const TokenParser = function () { };
+
+/**
+    A helper function to process and replace
+    all tokens within a string.
+*/
+TokenParser.replaceTokens = function (Str, Context) {
+    return TokenParser.parseTokens(TokenParser.extractTokens(Str), Context, Str);
+};
+
+/**
+    Will match all tokens "{some_token}, {some_token}
+    within a body of text, extracting all 
+    matches ( Tokens )
+    Will return an empty array, if no tokens are found.
+*/
+TokenParser.extractTokens = function (incomingString) {
+    var tokens = [];
+    incomingString.replace(/{(\w+)(}|\s{1})/g, (_, token) => {
+        tokens.push(token);
+    });
+    return tokens;
+};
+
+
+
+/**
+    Will loop through a set of tokens, replacing all matches within a 
+    string body, with the values supplied by the context. 
+*/
+TokenParser.parseTokens = function (Tokens, Context, incomingString) {
+    Tokens.forEach(function (token) {
+        if (token !== null && Context[token] !== undefined) {
+            incomingString = TokenParser.parseToken(token, Context[token], incomingString);
+        } else if (token !== null && Context[token] === undefined) {
+            incomingString = TokenParser.parseToken(token, '', incomingString);
+        }
+    });
+    return incomingString;
+};
+
+TokenParser.parseToken = function (Token, Value, Str) {
+    let regex = new RegExp("{(" + Token + ")}?", "g");
+    return Str.replace(regex, Value).replace('$', '');
```

Comment:
Can you replace the token "$" with "%" so as not to clash with js template literal

Reply:
So this one I don't like... `${FOO}` is a common placeholder string, so much so ES6 stole it. My next fallback is normal templating `{{FOO}}`.  `%{foo}` is odd -- if anything it should be `%FOO%` like DOS and other things that use % as vars. I think the linter in this case is dumb. 

So my preferences:
1. `${FOO}`
2. `{{FOO}}`
3. `%FOO%`

@arcepaul thoughts?

Reply:
if turning off eslint warnings is not an option, then I'd prefer to use `{{FOO}}` format

Reply:
Didn't `{{FOO}}` also throw errors?

Reply:
Just pushed updated code - all checks are passed

Reply:
I would prefer `3`

Reply:
Do you have a reason for the preference @seyilonge-nci ? I mean, I added some words around my thinking.

Reply:
Just realized your preference was in order, and you had a fallback. I'm fine with normal templating as suggested.

Reply:
I definitely don't like %{FOO} looking at it.  I like the handlebars style {{F00}} that Kate has moved to.